### PR TITLE
Cmake option to enable/disable vulkan video backend

### PR DESCRIPTION
**Pull request by mazes-80**
Allows build system to skip vulkan video backend at compile time.
Useful for "weak" hardware were vulkan is unsupported

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -536,7 +537,13 @@ endif()
 #
 add_subdirectory(Externals/Bochs_disasm)
 add_subdirectory(Externals/cpp-optparse)
-add_subdirectory(Externals/glslang)
+if(NOT APPLE)
```
Comment (riking) [inline CMakeLists.txt]: I don't think this is correct, to only enable Vulkan on Mac. This kind of force disabling is only used when the option definitely won't work.

Comment (mazes-80) [reply CMakeLists.txt]: It's already the way it's done in the current build system.
Vulkan is always compiled when not on an apple system.

Comment (mazes-80) [reply CMakeLists.txt]: Ah sorry, I get it the test has to be if(APPLE), not if(NOT APPLE)

Comment (mazes-80) [reply CMakeLists.txt]: Seems the option definitely won't work on apple.

**Comment by jordan-woyak** [comment]
Can you rebase this, please?

**Comment by mazes-80** [comment]
Rebase done

File: CMakeLists.txt [hunk 2]
```diff
@@ -543,8 +545,14 @@ else()
   message(STATUS "Using static fmt from Externals")
   add_subdirectory(Externals/fmt EXCLUDE_FROM_ALL)
 endif()
-add_subdirectory(Externals/glslang)
 add_subdirectory(Externals/imgui)
+if(APPLE)
+	set(ENABLE_VULKAN 0)
+endif()
```
Comment (spotlightishere) [inline CMakeLists.txt:551]: Support for macOS and Vulkan via MoltenVK was added after this PR was made (Nov. 2019, see 673f1963a068fb7d5cd8af9bb209c7633950e8f8) so the two `!defined(__APPLE__)`s below should be able to be safely removed! I tested this removal locally to confirm things build sanely.

**Comment by JosJuice** [comment]
A new function `VideoBackendBase::GetDefaultBackendName` was added to Dolphin very recently. It should be updated in a similar way to how you updated `VideoBackendBase::PopulateList`.

**Review by iwubcode — APPROVED** [review]

**Pull request merged**